exporter/metrics.go

```diff
@@ -35,26 +127,108 @@ func makeRawMetric(name string, value interface{}) (prometheus.Metric, error) {
 		f = float64(v)
 	case int64:
 		f = float64(v)
+	case float32:
+		f = float64(v)
 	case float64:
 		f = v
-	case primitive.A, primitive.ObjectID: // TODO: Fix this. returning nil, nil just to make tests pass
-		return nil, nil
-
 	case primitive.DateTime:
 		f = float64(v)
-	case primitive.Timestamp:
-		return nil, nil
-
-	case string:
+	case primitive.A, primitive.ObjectID, primitive.Timestamp, string:
 		return nil, nil
-
 	default:
 		return nil, fmt.Errorf("makeRawMetric: unhandled type %T", v)
 	}
 
-	fqName := makeRawMetricName(name)
-	help := "TODO"
+	if labels == nil {
+		labels = map[string]string{}
+	}
+
+	help := metricHelp(prefix)
 	typ := prometheus.UntypedValue
-	d := prometheus.NewDesc(fqName, help, nil, nil)
-	return prometheus.NewConstMetric(d, typ, f)
+
+	fqName, label := nameAndLabel(prefix, name)
+	if label != "" {
+		labels[label] = name
+	}
+
+	ln := make([]string, 0, len(labels))
+	lv := make([]string, 0, len(labels))
+
+	for k, v := range labels {
+		ln = append(ln, k)
+		lv = append(lv, v)
+	}
+
+	d := prometheus.NewDesc(fqName, help, ln, nil)
+
+	m, err := prometheus.NewConstMetric(d, typ, f, lv...)
+
+	return m, err
+}
+
+// metricHelp builds the metric help.
+// It is a very very very simple function, but the idea is if the future we want
+// to improve the help somehow, there is only one place to change it for the real
+// functions and for all the tests.
+func metricHelp(prefix string) string {
+	return prefix
+}
+
+func (d *diagnosticDataCollector) makeMetrics(prefix string, m bson.M, labels map[string]string) []prometheus.Metric {
```

Comment:
why it's still methods? Do we use anything from *diagnosticDataCollector?

Reply:
Made them independent functions